docs/Data_Submission_Portal/Users_Guide/Troubleshooting.md

```diff
@@ -0,0 +1,1236 @@
+  "code": 400,
+  "created_entity_count": 0,
+  "entities":
+    [
+        {
+            "action":null,
+            "errors":
+                [
+                    {
+                        "keys":["type"],
+                        "message":"missing 'type'",
+                        "type":"INVALID_TYPE"
+                    },
+                    {
+                        "keys":["type"],
+                        "message":"'type' is a required property",
+                        "type":"MISSING_PROPERTY"
+                    }
+                ]
+            "id":
+                [
+                    {
+                        "id":null
+                    }
+                ],
+            "related_cases":[],
+            "type":null,
+            "unique_keys":[],
+            "valid":false,
+            "warnings":[]
+        }
+    ],
+    "entity_error_count":1,
+    "message":"Transaction aborted due to 1 invalid entity.",
+    "success":false,
+    "transaction_id":6372998,
+    "transactional_error_count":0,
+    "transactional_errors":[],
+    "updated_entity_count":0
+}
+```
+```Request2
+{
+  "projects": {
+    "code": "INTERNAL"
+  }, 
+  "disease_type": "Adenomas and Adenocarcinomas", 
+  "primary_site": "Bladder",
+  "project_id": "GDC-INTERNAL", 
+  "type": "case",
+  "submitter_id": "GDC-INTERNAL-000099"
+}
+```
+```Response1
+{
+  "cases_related_to_created_entities_count": 0,
+  "cases_related_to_updated_entities_count": 0,
+  "code": 200,
+  "created_entity_count": 0,
+  "entities":
+    [
+        {
+            "action":"update",
+            "errors":[],
+            "id":
+                [
+                    {
+                        "id":"f0c68d72-ee54-4ad9-89e8-b08398b5aa0b"
+                    }
+                ],
+            "related_cases":[],
+            "type":"case",
+            "unique_keys":
+                [
+                    {
+                        "project_id":"GDC-INTERNAL",
+                        "submitter_id":"GDC-INTERNAL-000099"
+                    }
+                ],
+            "valid":true,
+            "warnings":[]
+        }
+    ],
+    "entity_error_count":0,
+    "message":"Transaction successful.",
+    "success":true,
+    "transaction_id":6372999,
+    "transactional_error_count":0,
+    "transactional_errors":[],
+    "updated_entity_count":1
+}
+```
+
+#### INVALID_VALUE Messages
+
+|Message|Explanation|Solution|
+| --- | --- | --- |
+| __None is not of type 'string'__ | If a property accepts string values, null is not an accepted value | Update the entity to have a string value for the given property, change null to a string "null", or remove the property if it is not required |
+| __{value} is not valid under any of the given schemas: {value} is less than the minimum of {minimum value} and {value} is not of type 'null'__ | The amount does not fall within the accepted minimum and maximum values | Ensure the value falls within the accepted range |
```

Comment:
Out of curiosity, do you know how to get this message vs. the similar one in the "ERROR" section?  No need to change anything.

Reply:
This has to do with whether the accepted value is "integer" or whether it's "One of number or null". If it's integer, the message in the ERROR section is generated, and if it's either a number or null then the one above is generated

Reply:
@wwysoc2 This has to do with whether the accepted value is "integer" or whether it's "One of number or null". If it's integer, the message in the ERROR section is generated, and if it's either a number or null then the one above is generated